src/python/bcc/table.py

```diff
@@ -421,31 +493,30 @@ def items_lookup_batch(self):
         if (res != 0 and errcode != errno.ENOENT):
             raise Exception("BPF_MAP_LOOKUP_BATCH has failed")
 
-        for i in range(0, count.value):
-            yield (keys[i], values[i])
-
-    def items_delete_batch(self, keys=None):
-        """Delete all the key-value pairs in the map if no key are given.
-        In that case, it is faster to call lib.bpf_lookup_and_delete_batch than
-        create keys list and then call lib.bpf_delete_batch on these keys.
-        If the array of keys is given then it deletes the related key-value.
+        for i in range(0, ct_cnt.value):
+            yield (ct_keys[i], ct_values[i])
+
+    def items_delete_batch(self, ct_keys=None):
+        """Delete the key-value pairs related to the keys given as parameters.
+        Note that if no key are given, it is faster to call
+        lib.bpf_lookup_and_delete_batch than create keys array and then call
+        lib.bpf_delete_batch on these keys.
+
+        Args:
+            ct_keys (ct.Array): keys array to delete. If an array of keys is
+            given then it deletes all the related keys-values.
+            If keys is None (default) then it deletes every entries.
+        Yields:
+            tuple: The tuple of (key,value) for every entries that have
+            been deleted.
+        Raises:
+            Exception: If bpf syscall has failed or is invalid
         """
-        if keys is not None:
-            # a ct.Array is expected
-            if not isinstance(keys, ct.Array):
-                raise TypeError
-
-            batch_size = len(keys)
-
-            # check that batch between limits and coherent with the provided values
-            if batch_size < 1 or batch_size > self.max_entries:
-                raise KeyError
-
-            count = ct.c_uint32(batch_size)
-
+        if ct_keys is not None:
+            ct_cnt = self._sanity_check_keys_values(keys=ct_keys)
             res = lib.bpf_delete_batch(self.map_fd,
-                                       ct.byref(keys),
-                                       ct.byref(count)
+                                       ct.byref(ct_keys),
+                                       ct.byref(ct_cnt)
                                        )
             errcode = ct.get_errno()
             if (errcode == errno.EINVAL):
```

Comment:
The error chacking here and below is not quite right. The delete_batch and update_batch use kernel internal function `generic_map_{delete,update}_batch` function, they return 0 to indicate a success and non-0 as an error code. Just check errcode is 0 or not is enough.